pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -32,26 +35,31 @@ part 'c_compiler_config.dart';
 /// be automatically run, by the Flutter and Dart SDK tools. The hook will be
 /// run with specific commandline arguments, which [BuildConfig] can parse and
 /// provide more convenient access to.
-abstract final class BuildConfig {
+abstract final class BuildConfig implements HookConfig {
```

Comment:
Can we drop the repeated APIs if we use `extends` instead of `implements`? (Fine with either solution.)